docs/proposals/31-inplace-node-update.md

```diff
@@ -236,43 +261,59 @@ metadata:
 
-Currently, when a dependency changes in the Operating System Configuration (OSC), `gardener-node-agent` detects the difference between the current OSC and the last applied OSC, and updates any changed unit on the node. When the in-place update strategy is configured, and either the Kubernetes minor version or the machine image version is changed, `gardener-node-agent` will not immediately apply the updates to the node. Instead, it will wait for the `ready-to-update` label on the node. Once the label is present, `gardener-node-agent` will proceed with the update.
+For OS updates, the OS extensions are responsible for populating the `inPlaceUpdateConfig` field within the OperatingSystemConfig status. This configuration includes the path to the update script (`osUpdateScriptPath`) and any additional arguments (`osUpdateScriptArgs`) required by the script. `gardener-node-agent` triggers the script located at the path specified in `inPlaceUpdateConfig.osUpdateScriptPath` and passes the arguments defined in `inPlaceUpdateConfig.osUpdateScriptArgs`.
 
-In the case of OS updates, `gardener-node-agent` expects an "update-script" to be present in the `.spec.files[]` of the OSC at a predefined path, and the same will be executed to perform the update.
+For Kubelet minor version or configuration updates, `gardener-node-agent` will apply the changed files and restart the `kubelet` unit and there are no additional steps involved.
 
-After the update, based on its success or failure, `gardener-node-agent` will add the appropriate label(`update-successful` or `update-failed`) to the node to indicate the status of the update.
+For credential rotation, in the case of Certificate Authority (CA) rotation, the `kubelet` needs to be bootstrapped again so that it can request client certificates signed by the new CA. `gardener-node-agent` will use the existing kubeconfig used by the `kubelet` (passed in the `--kubeconfig` flag. See [`kubelet` Options](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/)) and use it as bootstrap kubeconfig (`--bootstrap-kubeconfig` flag in the kubelet). It will remove the kubelet certificate directory (`--cert-dir` flag in the kubelet) `/var/lib/kubelet/pki`, and the `kubelet.service` unit is restarted. This makes the kubelet request new client certificates through [Certificate signing requests (CSR)](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#certificate-signing-requests).
+In case of `ServiceAccount` signing key rotation, `gardener-node-agent` will restart itself to load the new service account token using its [`token` controller](https://github.com/gardener/gardener/blob/master/docs/concepts/node-agent.md#token-controller).
```

Comment:
Is a restart really needed?

Reply:
Once upon a time, we had a watch. However, we removed it as part of https://github.com/gardener/gardener/pull/8857. However, since the node-agent knows when to re-read it (after performing the update), we could make it trigger the controller (via a channel source or so). This way, we can avoid the restart.

Reply:
Sure. Makes sense,